snapcraft/cli/remote.py

```diff
@@ -235,3 +208,27 @@ def _get_project_architectures(project) -> List[str]:
                 else:
                     archs.append(new_arch)
     return archs
+
+
+def _determine_architectures(project: Project, user_specified_arch: str):
+    # If build architectures not set in snapcraft.yaml, let the user override
+    # Launchpad defaults using --arch.
+    project_architectures = _get_project_architectures(project)
+    if project_architectures and user_specified_arch:
+        raise click.BadOptionUsage(
+            "Cannot use --arch, architecture list is already set in snapcraft.yaml."
+        )
+
+    if project_architectures:
+        archs = project_architectures
+    elif user_specified_arch == "all":
+        archs = _SUPPORTED_ARCHS
```

Comment:
your refactor made me notice this, please remove this option `all` has a very different meaning in the `architectures` list in `snapcraft.yaml`.

Reply:
Remove the `all` option altogether?

Reply:
Pulled out arch handling into its own PR (and fixed): https://github.com/snapcore/snapcraft/pull/2774/commits